BlueDB/src/main/java/org/bluedb/api/BlueQuery.java

```diff
@@ -3,24 +3,120 @@
 import java.io.Serializable;
 import java.util.List;
 import org.bluedb.api.exceptions.BlueDbException;
+import org.bluedb.api.keys.TimeFrameKey;
+import org.bluedb.api.keys.TimeKey;
 
 public interface BlueQuery<V extends Serializable> {
 
-	BlueQuery<V> where(Condition<V> c);
+	/**
+	 * Adds a condition to the query before returning itself.
+	 * @param condition - a filter function to be applied to possible matching values
+	 * @return itself, with the condition added to the query
+	 */
+	BlueQuery<V> where(Condition<V> condition);
 
+	/**
+	 * For queries on a collection with a key type of {@link TimeFrameKey}, this adds a condition that the value starts in the 
+	 * queried interval. Normally, it is only required that the value's time interval overlaps the queried time interval.
+	 * </br></br>
+	 * This can be very useful if you want to map reduce the results. Instead of making one query for the entire time interval you
+	 * can split it into multiple time intervals to be queried and processed individually. By using this condition on all but the 
+	 * first query you can guarantee that each value will only be processed once.
+	 * </br></br>
+	 * This condition only has an effect on collections with a key type of {@link TimeFrameKey}
+	 * 
+	 * @return itself, with the condition added to the query that the key start time be in the queried time interval
+	 */
 	BlueQuery<V> byStartTime();
 
+	/**
+	 * Adds an exclusive max time for the queried time interval. For a {@link TimeFrameKey} this means that it starts 
+	 * before the specified time. For a {@link TimeKey} it means that it is before the specified time. It doesn't make 
+	 * sense to use it with other key types.
+	 * 
+	 * @param time - an exclusive max time for the queried time interval
+	 * @return itself, with an exclusive max time for the queried time interval
+	 */
 	BlueQuery<V> beforeTime(long time);
+
+	/**
+	 * Adds an inclusive max time for the queried time interval. For a {@link TimeFrameKey} this means that it starts 
+	 * before or at the specified time. For a {@link TimeKey} it means that it is before or at the specified time. It doesn't 
+	 * make sense to use it with other key types.
+	 * 
+	 * @param time - an inclusive max time for the queried time interval
+	 * @return itself, with an inclusive max time for the queried time interval
+	 */
 	BlueQuery<V> beforeOrAtTime(long time);
 
+	/**
+	 * Adds an exclusive min time for the queried time interval. For a {@link TimeFrameKey} this means that it ends 
+	 * after the specified time. For a {@link TimeKey} it means that it is after the specified time. It doesn't make 
+	 * sense to use it with other key types.
+	 * 
+	 * @param time - an exclusive min time for the queried time interval

```

Comment:
For the time parameter in the time methods, we might want to mention that it's in ms since epoch